src/components/VButton.vue

```diff
@@ -2,7 +2,7 @@
   <Component
     :is="as"
     :type="typeRef"
-    class="flex appearance-none items-center justify-center rounded-sm no-underline ring-offset-1 transition-shadow duration-100 ease-linear focus:outline-none disabled:opacity-70"
+    class="inline-flex appearance-none items-center justify-center rounded-sm no-underline ring-offset-1 transition-shadow duration-100 ease-linear focus:outline-none disabled:opacity-70"
```

Comment:
Making the button `inline-flex` makes both `button` and `a` elements behave similarly. Without it, by default, an `<a>` button takes up all of the parent's width.